platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/address/AddressBookUtils.java

```diff
@@ -238,31 +251,38 @@ public static ServiceEndpoint endpointFor(@NonNull final String host, final int
+
+        if (addressBookInitializer.hasAddressBookChanged()) {
+            if (addressBookInitializer.getPreviousAddressBook() != null) {
+                // We cannot really "update" the previous roster because we don't know the round number
+                // at which it became active. And we shouldn't do that anyway because under normal circumstances
+                // the RosterService tracks the roster history correctly. However, since we're given a non-null
+                // previous AddressBook, and per the current implementation we know it comes from the state,
+                // we might as well validate this fact here just to ensure the update is correct.
+                final Roster previousRoster =
+                        RosterRetriever.buildRoster(addressBookInitializer.getPreviousAddressBook());
+                if (!previousRoster.equals(RosterRetriever.retrieveActiveOrGenesisRoster(state))
+                        && !previousRoster.equals(RosterRetriever.retrievePreviousRoster(state))) {
+                    throw new IllegalStateException(
+                            "The previousRoster in the AddressBookInitializer doesn't match either the active or previous roster in state."
+                                    + " AddressBookInitializer previousRoster = " + RosterUtils.toString(previousRoster)
+                                    + ", state currentRoster = "
+                                    + RosterUtils.toString(RosterRetriever.retrieveActiveOrGenesisRoster(state))
+                                    + ", state previousRoster = "
+                                    + RosterUtils.toString(RosterRetriever.retrievePreviousRoster(state)));
+                }
+            }
 
-        final boolean useRosterLifecycle = platformContext
-                .getConfiguration()
-                .getConfigData(AddressBookConfig.class)
-                .useRosterLifecycle();
-        if (!useRosterLifecycle && addressBookInitializer.hasAddressBookChanged()) {
-            final MerkleRoot state = initialState.get().getState();
-            // Update the address book with the current address book read from config.txt.
-            // Eventually we will not do this, and only transactions will be capable of
-            // modifying the address book.
-            final PlatformStateModifier platformState = state.getWritablePlatformState();
-            platformState.bulkUpdate(v -> {
-                v.setAddressBook(addressBookInitializer.getCurrentAddressBook().copy());
-                v.setPreviousAddressBook(
-                        addressBookInitializer.getPreviousAddressBook() == null
-                                ? null
-                                : addressBookInitializer
-                                        .getPreviousAddressBook()
-                                        .copy());
-            });
+            RosterUtils.setActiveRoster(
```

Comment:
If this code is committed, since this is not gated behind a feature flag, is this going to create a permanent switch to using rosters and not address books?     Does the app support this without having their feature flag turned on?